### PR TITLE
feat: implement basic auth workflow #1

### PR DESCRIPTION
**Pull request by boyeonihn**
- built login/logout feature utilizing useSupabaseClient & useSuperbaseUser composables 
- implement Google OAuth for login (also started github & email signin) -- for now will focus on one authentication method
- once auth flow is complete -> redirected to /confirm page, which reroutes to /today
- build signout button as reusable component

### CODE REVIEW

File: pages/login.vue
Comment (eunjae-lee) [inline pages/login.vue:71]: We cannot use `localhost` for production, so we need to do something about it. One way to deal with it is, to have an [environment variable](https://nuxt.com/docs/api/composables/use-runtime-config#environment-variables), like... 

```
MANYADAY_DOMAIN=localhost:3000
```

↑ The local `.env` file, and you can put the real domain (probably `something.vercel.app` at this point) in your hosting platform's env var configuration.

File: pages/today.vue
Comment (eunjae-lee) [inline pages/today.vue:4]: If you're not going to change this `date` value, no need to wrap it with `ref`. You could leave it as a static date variable.

File: supabase/db.sql
Comment (eunjae-lee) [inline supabase/db.sql:12]:
```suggestion
create policy "Can view own profile." on profiles for select using (auth.uid() = user_id);
```

**Review by eunjae-lee — APPROVED** [review]
Looks good!

It could be better if you could include some screenshots in the upcoming PRs, because it can help the reviewers to better contextualize and visualize the changes, and also it makes a good history of the project :)

**Comment by eunjae-lee** [comment]
This repo allows all options of merging PRs

Screenshot 2024-02-01 at 14 52 30

And I think you could go to Settings of this project, and keep only "squash merging" on. 

Screenshot 2024-02-01 at 14 52 53

The other options can be useful, but you could start with "squash merging" first and see if you need the others later.

File: pages/login.vue
Comment (boyeonihn) [reply pages/login.vue:71]: I see - to confirm, I will have to create a .env variable for the development mode url (localhost) and once I deploy to a real domain such as vercel, I input/configure the url on the platform settings -- is this correct?

**Comment by boyeonihn** [comment]
> ###1 approving review [Learn more about pull request reviews.](https://docs.github.com/articles/about-pull-request-reviews/)
> ** 1 approval **
> #### Require approval from specific reviewers before merging
> [Rulesets](https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets) ensure specific people approve pull requests before they're merged.
> 
> [Add rule](/boyeonihn/manyaday/settings/rules/new?target=branch&enforcement=disabled)
> 
> Show all checks
> ### All checks have passed
> 1 successful check
> ### This branch has no conflicts with the base branch
> Merging can be performed automatically.
> Merge pull request
> You can also [open this in GitHub Desktop](https://desktop.github.com) or view command line instructions .
> 
> [@boyeonihn](/boyeonihn)
> 
> 
> #### Add a comment
> Comment
> Write
> Preview
> 
> 
> Add your comment here...
> 
> [ Markdown is supported ](https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) 
> Paste, drop, or click to add files
>  Close pull request
> Comment
> Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines).
>  **ProTip!** Add comments to specific lines under [Files changed](/boyeonihn/manyaday/pull/5/files).
> Reviewers
> [ @eunjae-lee ](/eunjae-lee) [ eunjae-lee ](/eunjae-lee) [ ](/boyeonihn/manyaday/pull/5/files/4fcf63f7d296d3ecc0de12b9f62aa3062b81bb94) eunjae-lee approved these changes Re-request review
> 
> Still in progress?
> Convert to draft
> Assignees
> [ @boyeonihn ](/boyeonihn) [ boyeonihn ](/boyeonihn)
> 
> Labels
> None yet
> Projects
> None yet
> Milestone
> No milestone
> Development
> Successfully merging this pull request may close these issues.
> 
> None yet
> 
> Notifications
> Customize
> 
> Unsubscribe
> You’re receiving notifications because you’re watching this repository.
> 
> 2 participants
> [ @boyeonihn ](/boyeonihn) [ @eunjae-lee ](/eunjae-lee)
>  **Lock conversation**

Thank you for the suggestions!!! To confirm -- by screenshots do you mean screenshots of the app/the features that I added for this PR?

File: pages/login.vue
Comment (eunjae-lee) [reply pages/login.vue:71]: Yes! And now that I think again, MANYADSY_DOMAIN sounds like a wrong name. It could be 

MANYADAY_URL=http://localhost:3000

**Comment by eunjae-lee** [comment]
> Thank you for the suggestions!!! To confirm -- by screenshots do you mean screenshots of the app/the features that I added for this PR?

Yes!

File: pages/login.vue
Comment (boyeonihn) [reply pages/login.vue:71]: The redirect post google oauth wasn't working after I tried adding the baseURL to the `.env` file. I did a bit of digging and this is what I found: 

https://nuxt.com/docs/guide/going-further/runtime-config
CleanShot 2024-02-03 at 09 57 54@2x
- the docs talk about configuring the runtime configuration in the nuxt.config.ts file, so that's what I ended up trying and now everything seems to work. 

I also looked at the [sample todo nuxt/supabase project codebase](https://github.com/nuxt-modules/supabase/blob/main/demo/nuxt.config.ts) that I referred to for my sample nuxt project and they also did something similar to the docs mentioned above: 

CleanShot 2024-02-03 at 09 56 56@2x

Comment (eunjae-lee) [reply pages/login.vue:71]: Good job @boyeonihn !

Environment variables often include confidential informations, so frameworks like Nuxt or Next.js has their way to take the env vars into their system.

**Comment by eunjae-lee** [comment]
@boyeonihn after your PR is approved, if you add more commits, then you could click this button to re-request review.

Screenshot 2024-02-05 at 10 31 34

Or you could also leave a comment (tagging me), and mention what kind of changes you made, which makes the review easier.

**Comment by boyeonihn** [comment]
> @boyeonihn after your PR is approved, if you add more commits, then you could click this button to re-request review.
> 
> Screenshot 2024-02-05 at 10 31 34
> 
> Or you could also leave a comment (tagging me), and mention what kind of changes you made, which makes the review easier.

Duly noted!! Thank you so much ☺️

**Comment by eunjae-lee** [comment]
@boyeonihn I just found out something. You put `#1` as a part of the title here, and you can actually just write 

"fixes #1" or "closes #1" in the [PR description](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) (not necessarily in the title).

Then it will automatically closes the linked issue, but it doesn't necessarily include the issue number in the commit title.

Screenshot 2024-02-05 at 10 50 00

**Comment by boyeonihn** [comment]
> @boyeonihn I just found out something. You put `#1` as a part of the title here, and you can actually just write
> 
> "fixes #1" or "closes #1" in the [PR description](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) (not necessarily in the title).
> 
> Then it will automatically closes the linked issue, but it doesn't necessarily include the issue number in the commit title.
> 
> Screenshot 2024-02-05 at 10 50 00

@eunjae-lee oh great! I thought I had to put it in the PR title. Great information to know! :-)

**Pull request merged**
